photon-core/src/main/java/org/photonvision/common/configuration/NeuralNetworkModelManager.java

```diff
@@ -18,87 +18,270 @@
+     */
+    private NeuralNetworkModelManager() {}
 
+    /**
+     * Returns the singleton instance of the NeuralNetworkModelManager
+     *
+     * @return The singleton instance
+     */
     public static NeuralNetworkModelManager getInstance() {
         if (INSTANCE == null) {
             INSTANCE = new NeuralNetworkModelManager();
         }
         return INSTANCE;
     }
 
+    /** Logger for the NeuralNetworkModelManager */
+    private static final Logger logger = new Logger(NeuralNetworkModelManager.class, LogGroup.Config);
+
     /**
-     * Perform initial setup and extract default model from JAR to the filesystem
+     * Determines the model version based on the model's filename.
+     *
+     * <p>"yolov5" -> "YOLO_V5"
      *
-     * @param modelsFolder Where models live
+     * <p>"yolov8" -> "YOLO_V8"
+     *
+     * @param modelName The model's filename
+     * @return The model version
      */
-    public void initialize(File modelsFolder) {
-        var modelResourcePath = "/models/" + MODEL_NAME;
-        this.defaultModelFile = new File(modelsFolder, MODEL_NAME);
-        extractResource(modelResourcePath, defaultModelFile);
+    private static RknnJNI.ModelVersion getModelVersion(String modelName)
+            throws IllegalArgumentException {
+        if (modelName.contains("yolov5")) {
+            return RknnJNI.ModelVersion.YOLO_V5;
+        } else if (modelName.contains("yolov8")) {
+            return RknnJNI.ModelVersion.YOLO_V8;
+        } else {
+            throw new IllegalArgumentException("Unknown model version for model " + modelName);
+        }
+    }
+
+    /** This class represents a model that can be loaded by the RknnJNI. */
+    public class Model {
+        public final File modelFile;
+        public final RknnJNI.ModelVersion version;
+        public final List<String> labels;
+        public final Size inputSize;
+
+        /**
+         * Model constructor.
+         *
+         * @param model format `name-width-height-model.format`
+         * @param labels
+         * @throws IllegalArgumentException
+         */
+        public Model(String model, String labels) throws IllegalArgumentException {
+            this.modelFile = new File(model);
+
+            String[] parts = modelFile.getName().split("-");
+            if (parts.length != 4) {
+                throw new IllegalArgumentException("Invalid model file name: " + model);
+            }
+
+            // TODO: model 'version' need to be replaced the by the product of 'Version' x 'Format'
+            this.version = getModelVersion(parts[3]);
 
-        File labelsFile = new File(modelsFolder, "labels_v5.txt");
-        var labelResourcePath = "/models/" + labelsFile.getName();
-        extractResource(labelResourcePath, labelsFile);
+            int width = Integer.parseInt(parts[1]);
+            int height = Integer.parseInt(parts[2]);
+            this.inputSize = new Size(width, height);
```

Comment:
Double check if DNN lets you get model input image size from the model itself, but if it does, let's kill this part of the filename structure

Reply:
I agree! I'll look into adding that as part of the JNI.

Reply:
I can't find anything in the OpenCV DNN docs. https://docs.opencv.org/4.x/javadoc/org/opencv/dnn/package-summary.html